### PR TITLE
Fix #305: Fix PDF export from matplotlib

### LINKED ISSUE #305: The PDF ``button`` on the matplotlib widgets toolbar doesn't seem to work
The notebook for Pyodide available here: https://extremely-alpha.iodide.io/notebooks/151/ has a demo on how to use ``matplotlib`` to plot charts. There are 3 buttons which enable the user to save the chart in various formats - ``PDF, PNG and SVG``. 

Clicking on any of these should result in a pop-up window something like this:

screenshot 2019-01-25 at 12 37 01 pm

However, clicking on the ``PDF`` button does not result in a pop-up window. Instead, it embeds another chart inside the original chart something like this:

screenshot 2019-01-25 at 12 34 13 pm

I don't think this is the desired behaviour.

### PR DESCRIPTION
**Pull request by mdboom**
This was broken by the recent upgrade of emscripten, which was no longer including C++ exception support (required by matplotlib's PDF export).

This adds a test to help us catch this case again in the future.

### CODE REVIEW

File: test/test_matplotlib.py [hunk 1]
```diff
@@ -16,3 +16,15 @@ def test_svg(selenium):
...
+
+
+def test_pdf(selenium):
+    selenium.load_package("matplotlib")
+    selenium.run("from matplotlib import pyplot as plt")
+    selenium.run("plt.figure()")
+    selenium.run("x = plt.plot([1,2,3])")
+    selenium.run("import io")
+    selenium.run("fd = io.BytesIO()")
+    selenium.run("plt.savefig(fd, format='svg')")
```
Comment (madhur-tandon) [inline test/test_matplotlib.py]: Shouldn't the format be ``pdf`` instead of ``svg`` ?

**Comment by madhur-tandon** [comment]
@mdboom Sorry for the late review, I was out somewhere. With that said, I have expressed my initial thoughts and am currently building ``emsdk`` to use the new version of ``emscripten``. I will check the changes locally and then give my final thoughts. Does this suffice?

**Comment by madhur-tandon** [comment]
@mdboom I am done with my local testing and it works! So this PR should be ready to merge except the minor comment that I left above about the ``format`` in the test ``test_pdf``.

File: test/test_matplotlib.py (hunk 1 above)
Comment (mdboom) [reply test/test_matplotlib.py]: Good catch!  Thanks.

**Comment by madhur-tandon** [comment]
@mdboom The current test fails because the length of the content is not ``15752``, it is ``5559``. Infact, this is the reason for the tests failing on circleCI as well. Changing that should make the test ``test_pdf`` pass. We could then merge this finally :D

File: test/test_matplotlib.py [hunk 2]
```diff
@@ -16,3 +16,15 @@ def test_svg(selenium):
...
+def test_pdf(selenium):
+    selenium.load_package("matplotlib")
+    selenium.run("from matplotlib import pyplot as plt")
+    selenium.run("plt.figure()")
+    selenium.run("x = plt.plot([1,2,3])")
+    selenium.run("import io")
+    selenium.run("fd = io.BytesIO()")
+    selenium.run("plt.savefig(fd, format='pdf')")
+    content = selenium.run("fd.getvalue()")
+    assert len(content) == 15752
```
Comment (madhur-tandon) [inline test/test_matplotlib.py]: This should be ``5559`` instead.

Comment (mdboom) [reply test/test_matplotlib.py]: Thanks

**Pull request merged**